### DISCUSSION TITLE
Allow to check job permission inside a step

### DISCUSSION THREAD

**Discussion by 0ssigeno** [community/community#34309] [Actions]
Hi guys!
 I did not find any documentation about having the possibility to check, inside the job, with which permission the job itself is run (i.e. read, write, admin).
 
 Use case: I wrote a reusable workflow for our Org, and I want to execute a step (that for example it is writing a comment on the repository) iif the owner of the repository assigned the `write` or `admin` permission to the job.
 
 Is it already possible to check that? If not, there are any tricks that I can use to solve my use case?

**Comment by dgholz** [comment]
> Is it already possible to check that?

It's not.

> If not, there are any tricks that I can use to solve my use case?

Some other people have had similar questions, and wrote audit tools (like https://github.com/stoe/action-permissions-cli) to show the permissions on a workflow/job. You could probably use or adapt it?

But for your actual use case:

> I want to execute a step (that for example it is writing a comment on the repository) iif the owner of the repository assigned the write or admin permission to the job.

Usually people would specify what the action should do with an input, instead of the action being smart and implicitly taking actions. I think it's an interesting idea though!

**Reply by 0ssigeno** [reply]
Thanks for the answer!

I will check on that, but I don't think that the solution that I can write with it would be clean enough to actually deploy it in a production environment.

> Usually people would specify what the action should do with an input, instead of the action being smart and implicitly taking actions. 

I'm already doing that, meaning that the release step is triggered only if the repo owner configured to actually run it. What I wanted to achieve was, let me say this, have a better UX experience for the caller: if the permissions weren't correctly set, I could print/return an understandable error, while, if I remember correctly, if you try to  execute an action for which you don't have the permissions, the explanation is not so clear. On this last step I could absolutely be mistaken.
